numpy/core/src/multiarray/shape.c

```diff
@@ -680,9 +680,9 @@ PyArray_SwapAxes(PyArrayObject *ap, int a1, int a2)
 NPY_NO_EXPORT PyObject *
 PyArray_Transpose(PyArrayObject *ap, PyArray_Dims *permute)
```

Comment:
We're kinda abusing `PyArray_Dims`. This isn't a list of dims (`intp`), this is a list of axes (`int`).